src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -314,9 +315,7 @@ def create_managed_disk(resource_group_name, disk_name, location=None,
 
     if size_gb is None and option == DiskCreateOption.empty:
         raise CLIError('usage error: --size-gb required to create an empty disk')
-
-    disk = Disk(location, disk_size_gb=size_gb, creation_data=creation_data,
-                account_type=sku, tags=(tags or {}))
+    disk = Disk(location, creation_data, (tags or {}), _get_sku_object(sku), disk_size_gb=size_gb)
```

Comment:
I always try to use kwargs if I can; but in this specific case, I ended up with a big `if` block around the api-version to invoke the constructor mostly with the same arg list. The change here is a trade-off to get code a bit clean therefore readable, and also less code for api-version check as `snapshot create` can also use the same `_get_sku_object` . Sure, this makes code a bit brittle, i agree, but we have 2 good tests for this though.